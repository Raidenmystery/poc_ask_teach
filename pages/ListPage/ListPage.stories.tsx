import { ComponentStory, ComponentMeta } from '@storybook/react-native'
import { QueryClientProvider } from 'react-query';
import { queryClient } from '../../shared/config';
import ListPage from './index';

export default {
  title: "ListPage",
  component: ListPage,
} as ComponentMeta<typeof ListPage>;

const Template: ComponentStory<typeof ListPage> = () => (
  <QueryClientProvider client={queryClient}>
    <ListPage />
  </QueryClientProvider>
);

export const Default = Template.bind({});
