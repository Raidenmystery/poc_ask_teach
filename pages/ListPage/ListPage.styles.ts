import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  item: {
    backgroundColor: "#f2f2f2",
    padding: 20,
    marginVertical: 10,
    borderRadius: 10,
  },
  itemText: {
    fontSize: 18,
    fontWeight: "bold",
  },
  modal: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  modalText: {
    fontSize: 24,
    marginBottom: 20,
  },
  modalButton: {
    backgroundColor: "#2196F3",
    padding: 10,
    borderRadius: 5,
  },
  modalButtonText: {
    color: "#fff",
    fontSize: 18,
  },
});
