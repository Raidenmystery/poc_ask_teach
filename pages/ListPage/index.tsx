import React, { useState } from "react";
import { View, Text, TouchableOpacity, Modal } from "react-native";

import { styles } from "./ListPage.styles";
import { usePokemonData } from "../../hooks/queries";
import { NavigationButtons } from "../../src/templates";

export default function ListPage({ navigation }: any): JSX.Element {
  const [modalVisible, setModalVisible] = useState(false);

  const data = usePokemonData("pikachu");

  const togleeModal = (value: string) => {
    setModalVisible(!modalVisible);
  };

  const pokemons = [
    { property: "Base experience", value: data?.data?.base_experience },
    { property: "Base height", value: data?.data?.height },
    { property: "Order", value: data?.data?.order },
  ];

  return (
    <View style={styles.container}>
      {pokemons.map(({ property, value }) => (
        <TouchableOpacity
          onPress={() => togleeModal(value)}
          style={styles.item}
          key={property}>
          <Text style={styles.itemText}>{property}</Text>
        </TouchableOpacity>
      ))}
      <NavigationButtons navigation={navigation} type="List" />
      <Modal visible={modalVisible} animationType="slide">
        <View style={styles.modal}>
          <Text style={styles.modalText}>
            Base experience: {pokemons[0].value}
          </Text>
          <Text style={styles.modalText}>Base height: {pokemons[1].value}</Text>
          <Text style={styles.modalText}>Order: {pokemons[2].value}</Text>
          <TouchableOpacity
            onPress={() => setModalVisible(!modalVisible)}
            style={styles.modalButton}>
            <Text style={styles.modalButtonText}>Close Modal</Text>
          </TouchableOpacity>
        </View>
      </Modal>
    </View>
  );
}
