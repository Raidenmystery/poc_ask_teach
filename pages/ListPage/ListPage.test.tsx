import { QueryClient, QueryClientProvider } from "react-query";
import renderer from "react-test-renderer";

import ListPage from "../ListPage";

const queryClient = new QueryClient();

describe("<ListPage />", () => {
  it("has 4 child", () => {
    const tree = renderer.create(
    <QueryClientProvider client={queryClient}>
      <ListPage />
    </QueryClientProvider>
    )
    .toJSON();
    expect(tree?.children?.length).toBe(4);
  });
  it("renders correctly", () => {
    const tree = renderer.create(
      <QueryClientProvider client={queryClient}>
        <ListPage />
      </QueryClientProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});