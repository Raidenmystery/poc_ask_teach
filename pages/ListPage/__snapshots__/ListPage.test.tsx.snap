// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListPage /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#f2f2f2",
        "borderRadius": 10,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Base experience
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#f2f2f2",
        "borderRadius": 10,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Base height
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#f2f2f2",
        "borderRadius": 10,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Order
    </Text>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={false}
  />
</View>
`;
